PartitioningAgent/Agent.cs

```diff
@@ -0,0 +1,58 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.Threading.Tasks;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Clustering;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Concurrency;
+using Microsoft.Azure.IoTSolutions.DeviceSimulation.Services.Diagnostics;
+
+namespace Microsoft.Azure.IoTSolutions.DeviceSimulation.PartitioningAgent
+{
+    public interface IPartitioningAgent
+    {
+        Task StartAsync();
+        void Stop();
+    }
+
+    public class Agent : IPartitioningAgent
+    {
+        private readonly IClusterNodes clusterNodes;
+        private readonly IThreadWrapper thread;
+        private readonly ILogger log;
+        private readonly int checkIntervalMsecs;
+
+        private bool running;
+
+        public Agent(
+            IClusterNodes cluster,
+            IThreadWrapper thread,
+            IClusteringConfig clusteringConfig,
+            ILogger logger)
+        {
+            this.clusterNodes = cluster;
```

Comment:
Where does this get passed in from? If Autofac is doing it, I'm not clear on where the values are coming from.

Reply:
I don't understand the question, `cluster` is injected as usual, see line 26